app/models/katello/host/subscription_aspect.rb

```diff
@@ -51,6 +53,31 @@ def self.new_interfaces(facts)
         end
       end
 
+      def self.find_or_create_host(organization, name, rhsm_params)
+        hosts = ::Host.where(:name => name)
+        if hosts.empty? #no host exists
+          Katello::Host::SubscriptionAspect.new_host_from_rhsm_params(rhsm_params, organization, Location.default_location)
+        elsif hosts.where(:organization_id => organization.id).empty? #not in the correct org
+          #TODO
+          fail "Can't handle registering to host in a different org, need to handle this case."
+        else
+          hosts.first
+        end
+      end
+
+      def self.find_or_create_host_for_hypervisor(name, organization, location = nil)
```

Comment:
This function and `find_or_create_host` share a lot of the same code -- could they be re-factored?